docs/hello-neighbour.yaml

```diff
@@ -184,212 +224,130 @@ paths:
-      responses:
-        200:
-          description: "Question updated"
-          schema:
-            $ref: "#/definitions/Question"
-        404:
-          description: "Question not found"
-          schema:
-            $ref: "#/definitions/ErrorResponse"
-        400:
-          description: "Invalid input"
-          schema:
-            $ref: "#/definitions/ErrorResponse"
-    delete:
-      tags:
-      - "leader"
-      summary: "Delete question"
-      description: "Delete a question from the system."
-      operationId: "deleteQuestion"
+      summary: "Retrieve list of all your church match groups"
+      description: "Retrieve all match groups of a specific church (leader) or the ones you are currently part of (member)."
+      operationId: "getMatchGroups"
       produces:
       - "application/json"
       parameters:
-      - in: "path"
-        name: "churchUuid"
-        type: "string"
-        format: "uuid"
-        required: true
-      - in: "path"
-        name: "questionUuid"
-        type: "string"
-        format: "uuid"
+      - name: "churchUuid"
+        in: "path"
         required: true
-      responses:
-        204:
-          description: "Question deleted"
-        404:
-          description: "Question not found"
-          schema:
-            $ref: "#/definitions/ErrorResponse"
-
-  /church/matchgroups: # church and user are determined based on user session
-    get:
-      tags:
-      - "leader"
-      summary: "Retrieve list of all your church match groups"
-      description: "Retrieve all match groups of a specific church."
-      operationId: "getMatchGroups"
-      parameters:
-      - in: "path"
-        name: "churchUuid"
         type: "string"
         format: "uuid"
-        required: true
       responses:
         200:
           description: "Match group list"
           schema:
             type: "array"
             items:
               $ref: "#/definitions/Group"
+        400:
+          $ref: "#/responses/ClientError"
+        401:
+          $ref: "#/responses/Unauthenticated"
         404:
           description: "Church not found"
           schema:
-            $ref:  "#/definitions/ErrorResponse"
-            
+            $ref:  "#/definitions/Error"
+
   /user:
     post:
       tags:
-      - "member"
-      summary: "Create new user account"
-      description: "Create a new user to be authenticated via Firebase. Both church leaders and members need such an account."
-      operationId: "createUser"
+      - "administrator"
+      summary: "Create new user"
+      description: "Add a new user to the system."
+      operationId: "addUser"
       consumes:
       - "application/json"
       produces:
       - "application/json"
       parameters:
       - in: "body"
         name: "body"
-        description: "User details for new account"
+        description: "User details"
         required: true
         schema:
           $ref: "#/definitions/UserIn"
```

Comment:
Isn't this the same as the `/register` endpoint? Couldn't the admin user just use that?

Reply:
We might need to reduce the `/register` endpoint to only take the fields editable upon invite. Otherwise, it is unclear what is accepted under what circumstances.

Reply:
I created a separate endpoint for members to accept invites: `/register/{userUuid}`. The only field that would not need to be editable would be `email`, but it is already tied to the UUID that we've generated, so it shouldn't be an issue.
My initial question was whether the Admin can just use the default `/register` instead of `/user` as they are the same.

Reply:
We could also turn that around and just say that registration is `PUT /user/{userUuid}` which you would be authorised to do based on just knowing the ID if the account hasn't been activated, yet, Then, we can just do away with the `/register` endpoint.

Reply:
The current `/user/{userUuid}` is currently for updating user information - and it requires authentication access. However, the 2 `/register` endpoints do not require authentication, so we'll still have to leave them in.